src/plib/library/Dns.php

```diff
@@ -34,4 +34,24 @@ public static function txtRecords($domainName)
             }, $dnsRecords)
         );
     }
+
+    /**
+     * @param $domainName
+     * @return mixed
+     * @throws \pm_Exception
+     */
+    public static function mxRecord($domainName)
+    {
+        $dnsRecord = @dns_get_record($domainName, DNS_MX);
+        if (false === $dnsRecord) {
+            $error = error_get_last()['message'];
+            throw new \pm_Exception("Failed to resolve {$domainName}: {$error}");
+        }
+
+        if (isset($dnsRecord[0]['host'])) {
+            return ['host' => $dnsRecord[0]['host'], 'ip' => self::aRecord($dnsRecord[0]['host']), 'priority' => $dnsRecord[0]['pri']];
```

Comment:
I think `host` could be also IP address for MX record - in this case, the resolve of A record fails

Reply:
According to 
https://tools.ietf.org/html/rfc1035#section-3.3.9
MX must contain <domain-name>